algorithms/a3c/env_vizdoom.py

```diff
@@ -0,0 +1,379 @@
+"""
+Adapted from: https://github.com/devendrachaplot/DeepRL-Grounding/blob/master/env.py
+
+This countains the GroundingEnv() with some additions to make it work with the ai2thor and atari
+code e.g. the small Task class because the real work was kept
+"""
+
+from __future__ import print_function
+from time import sleep
+import collections
+import codecs
+import json
+import warnings
+import os
+
+import numpy as np
+from gym import spaces
+
+from algorithms.a3c.vizdoom_utils.doom import *   # todo probably avoid this but at least it is self-contained within this file
+from algorithms.a3c.vizdoom_utils.points import *
+from algorithms.a3c.vizdoom_utils.constants import *
+
+actions = [[True, False, False], [False, True, False], [False, False, True]]
+
+ObjectLocation = collections.namedtuple("ObjectLocation", ["x", "y"])
+AgentLocation = collections.namedtuple("AgentLocation", ["x", "y", "theta"])
+
+
+class Task:
```

Comment:
It might be a bit risky since this is a different meaning of task and I purely wanted to just make it work with ViZDoom and not focus too long on making ViZDoom and our task interface similar. 

In other words, the functionality of the true task of ViZDoom environment is within GroundingEnv, not that fake task class just so I can have that variable. Feel free to argue.

Reply:
Yeah but since it is easy to abstract and then makes users' life easier, I'd say we do it. You just have to put the get_reward in a task and few little things more. I think it is totally worth it

Reply:
Do you still think I should do this? Moving vizdoom specific code into ai2thor tasks.py is risky and hard to do for a few reasons. Why try to make everything fit into our tasks structure, especially since I didn't write this code?